internal/stack/serverlessresources.go

```diff
@@ -65,3 +84,16 @@ func applyServerlessResources(profile *profile.Profile, stackVersion string, con
 
 	return nil
 }
+
+// esHostWithPort checks if the es host has a port already added in the string , else adds 443
+// This is to mitigate a known issue in logstash - https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-serverless
+func esHostWithPort(host string) string {
+	// The ES host of the form https://esHost or https://esHost:port
+	splitHost := strings.Split(host, ":")
+
+	// https://esHost:port port already defined
+	if len(splitHost) > 2 {
+		return host
+	}
+	return host + ":443"
```

Comment:
This code will probably not work as expected with IPv6 and maybe under other scenarios where splitting or joining host and port is not so straightforward.

Use `net.SplitHostPort` and `net.JoinHostPort`. Though probably `url.Parse` is better if we expect full URLs here.

Reply:
I think I can still use `url.Parse`. Because in serverless, the es_host comes with scheme and hostname.

For Ipv6

From [RFC 3986](http://www.ietf.org/rfc/rfc3986.txt), section 3.2.2: Host

>A host identified by an Internet Protocol literal address, version 6 [RFC3513] or later, is distinguished by enclosing the IP literal within square brackets ("[" and "]").


Added tests for these

Reply:
Looks good, thanks, added another case we might have.